plugins/modules/sssd_info.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/python
+# Copyright (c) 2025 Aleksandr Gabidullin <qualittv@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import annotations
+
+
+DOCUMENTATION = r"""
+---
+module: sssd_info
+version_added: 12.2.0
+short_description: Check SSSD domain status using D-Bus
+description:
+    - Check the online status of SSSD domains, list domains, and retrieve active servers using D-Bus.
+author: "Aleksandr Gabidullin (@MikeyTide)"
+requirements:
+    - dbus
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: none
+    platform:
+        platforms: posix
+        description: This action requires a system with D-Bus and SSSD running.
+        support: full
+options:
+    action:
+        description:
+            - The action to perform.
+            - V(domain_status) - Check if domain is online.
+            - V(domain_list) - List all configured domains.
+            - V(active_servers) - Get active servers for domain.
+            - V(list_servers) - List all servers for domain.
+        type: str
+        required: true
+        choices: ['domain_status', 'domain_list', 'active_servers', 'list_servers']
+    domain:
+        description:
+            - Domain name to check.
+            - Required unless O(action=domain_list).
+            - When O(action=domain_list), this parameter is ignored and the module returns a list of all configured domains.
+        type: str
+    server_type:
+        description:
+            - Required parameter when O(action=active_servers) and O(action=list_servers).
+            - Optional and ignored for all other actions.
+            - Valid values depend on your SSSD configuration, but typically include C(IPA), C(AD).
+            - This determines which type of authentication/identity servers to query (e.g., IPA servers for FreeIPA domains).
+        type: str
+        choices: ['IPA', 'AD']
```

Comment:
As you mentioned above that this is a free-form config (depends on the SSSD configuration), we should not use `choices` here, because it will restrict this parameter to these two options here.

Maybe in the future you might want to add an action to this module to retrieve the valid server_types for one particular SSSD configuration, but unless that is extremely trivial, I would leave it for another PR.

Reply:
As long as you remove the `choices` attribute, from here and from the argument_spec down below, that is OK.